web/src/js/components/contentviews/HttpMessage.tsx

```diff
@@ -60,6 +63,28 @@ export default function HttpMessage({ flow, message }: HttpMessageProps) {
         }
     }, [content]);
 
+    const handleClickCopyButton = () => {
+        const url = MessageUtils.getContentURL(flow, message, contentView); //without the 'maxLines' parameter, so we can get the full content of the content view
+
+        fetchApi(url)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error(
+                        `${response.status} ${response.statusText}`.trim(),
+                    );
+                }
+                return response.json();
+            })
+            .then((data: ContentViewData) => {
+                setIsCopied(true);
+                copyViewContentDataToClipboard(data);
+                setTimeout(() => setIsCopied(false), 10000);
```

Comment:
10 seconds feels a little excessive for a short user feedback. Does it reset when switching to a different flow? Maybe it's enough if it flashes `Copied!` for a short time?

Depending on if the component is re-used for multiple flows then `setTimeout` also needs to be cleared otherwise if I copied multiple flows in sequence you have multiple `setTimeout` racing and `isCopied` loses any meaning.

I'd remove the `disabled={isCopied}` part. There's no reason to limit the user in this way. If I want to copy the content again, let me.

Reply:
What you could do is `disabled={isFetching}`, so that when a big response is fetched the user can not click the button multiple times during the fetch (because there might not be immediate user feedback that the click did something).

> I didn't want the user to abuse it and make too many unnecessary calls

I just checked and mitmproxy serves the requests with ETag. If the user click the button again after the first request was made, the content should come from the browser cache. Can you verify this with the copy button as well?

Reply:
Yeah, there is an ETag, but it doesn't look like the response is cached. It seems that a new request is being made. The request header includes: `cache-control: no-cache`

Reply:
You probably have cache disabled in dev tools network tab?

Reply:
yes @Prinzhorn, you are right! I forgot to enable caching. Everything works as expected then :)